pcsx2/ps2/PGPU.cpp

```diff
@@ -0,0 +1,1110 @@
+	Console.WriteLn( "PGPU DMA read  0x%08X = 0x%08X",  addr, data);
+	return data;
+}
+
+void psxDma2GpuW(u32 addr, u32 data) {
+	#if LOG_PGPU_DMA == 1
+	Console.WriteLn( "PGPU DMA write 0x%08X = 0x%08X",  addr, data);
+	#endif
+	addr &= 0x1FFFFFFF;
+	if (addr == PGPU_DMA_MADR) {
+		pgpuDmaMadr = (data & 0x00FFFFFF);
+	} else if (addr == PGPU_DMA_BCR) {
+		pgpuDmaBcr = data;
+	} else if (addr == PGPU_DMA_CHCR) {
+		pgpuDmaChcr = data;
+		if (pgpuDmaChcr & DMA_START_BUSY) {
+			processPgpuDma();
+		}
+	} else if (addr == PGPU_DMA_TADR) {
+		pgpuDmaTadr = data;
+		Console.WriteLn( "PGPU DMA read TADR !!! ");
+	}
+	return;
+}
+
+
+
+
+
+//#########################################################################################################
+//Other debug & test code, not directly relatedto the PGIF:
+
+//-------------------------
+//debug print from IOP:
+
+//BIOS POST
+void ps12PostOut(u32 mem, u8 value) {
+	Console.WriteLn( "XXXXXXXXXXXXXXXXXXXXXXX PS12 POST = %02X ", value);
+	//iopMemWrite32(0x0000B9B0, 1); //a bit of high-level emulation to enable DUART TTY in PS1 kernel...which doesn't work
+	return;
+}
+
+
+void psDuartW(u32 mem, u8 value) {
+	Console.WriteLn( "XXXXXXXXXXXXXXXXXXXXXXX DUART reg %08X = %02X >%c<", mem, value, value);
+	if ((mem & 0x1FFFFFFF) == 0x1F802023) {
+		Console.WriteLn( "XXXXXXXXXXXXXXXXXXXXXXX DUART A = %02X >%c<", value, value);
+	} else if ((mem & 0x1FFFFFFF) == 0x1F80202B) {
+		Console.WriteLn( "XXXXXXXXXXXXXXXXXXXXXXX DUART B = %02X >%c<", value, value);
+	}
+	return;
+}
+
+u8 psExp2R8(u32 mem) {
+	if ((mem & 0x1FFFFFFF) == 0x1F802000) {
+		return 0x18;
+	}
+	return 0xFF;
+}
+
+int flgint = 0;
+
+char strb[200] = {0};
+void kernelTTYFileDescrWrite(u32 mem, u32 data) {
+	//data is an address actually
+	char stra[100];
+	int i, k;
+//	if (mem == ) {
+		k = iopMemRead32(data);
+
+		for (i=0; i<100; i++) stra[i] = iopMemRead8(data + i);
+		stra[99] = 0;
+
+		//Console.WriteLn("Wr to PS1 kernel file TTY descr. data = %08X  at addr %08X >%s< ", data, k, stra);
+
+		//Console.WriteLn("%s", stra);
+
+
+		if ((strlen(strb) + strlen(stra)) >= 200) {
+			//Console.WriteLn("%s", strb);
+			strb[0] = 0;
+		} else {
+
+			strcat(strb, stra);
+			if ((stra[strlen(stra) - 1] == 0xA) || (stra[strlen(stra) - 1] == 0xD) || (stra[strlen(stra) - 2] == 0xA) || (stra[strlen(stra) - 2] == 0xD)) {
+				Console.WriteLn(">%s<", strb);
+				if (strcmp(strb, "VSync") == 0) flgint ^=1;
+				if (strncmp(strb, "Execute", 5) == 0) { Console.WriteLn(" ###EXecute was just printed####");   doAnyIopLs =1;
+				}
+				strb[0] = 0;
+			}
+		}
+	//}
+}
+
+
+//0x80051080
+u32 getIntTmrKReg(u32 mem, u32 data) {
+	u32 outd = data;
+	if (PGifCtrlReg != 0) {
+		Console.WriteLn("XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX");
+		//outd = 0;
+		outd = 0x7FFFFFFF;
+	}
+	return outd;
+}
+
+int cntint = 0;
+void testInt(void) {
+
+
+	if (flgint == 1) {
+		if (cntint > 500) {
+	//Console.WriteLn("  ZZZZZZZZZZZZZZZZ  cpuEventTest_Shared  ");
```

Comment:
The entire testInt() appears to be an early debug leftover. It can go.
In R5900.cpp, the call to testInt() is also the only reason for the PGPU.h include. So both can be removed.